js/data/buffer.js

```diff
@@ -47,16 +47,22 @@ function Buffer(options) {
         var attributeAlignment = this.type === Buffer.BufferType.VERTEX ? Buffer.VERTEX_ATTRIBUTE_ALIGNMENT : 1;
 
         for (var i = 0; i < options.attributes.length; i++) {
-            var attribute = util.extend({}, options.attributes[i]);
-            this.attributes.push(attribute);
+            var attribute = util.extend({
+                components: 1,
+                type: Buffer.AttributeType.UNSIGNED_BYTE
+            }, options.attributes[i]);
+
+            assert(attribute.type.name in Buffer.AttributeType);
 
-            attribute.name = attribute.name;
-            attribute.components = attribute.components || 1;
-            attribute.type = attribute.type || Buffer.AttributeType.UNSIGNED_BYTE;
             attribute.size = attribute.type.size * attribute.components;
             attribute.offset = this.itemSize;
 
+            assert(!isNaN(attribute.size));
+
+            this.attributes.push(attribute);
+
             this.itemSize = align(attribute.offset + attribute.size, attributeAlignment);
+            assert(!isNaN(this.itemSize));
```

Comment:
I think this constructor was more clear the way it was written before. It was nice to have all the attribute object's fields initialized the same fashion.